### PR TITLE
Add syntax coloring for group by and collect clauses

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Test results:
Screenshot 2023-05-16 at 14 59 23

Final coloring output:
Screenshot 2023-05-16 at 14 56 49

### CODE REVIEW

File: test/resources/config/query.bal [hunk 1]
```diff
@@ -30,4 +30,12 @@ function testFunction() {
             where memberAge > minMemberAge
             select memberAge).pop() >  minLeaderAge
         select leader;
+
+    var orders = [{orderCategory: "A", price: 123, id: 1}, {orderCategory: "B", price: 123, id: 2}];
+
+    var _  = from var {orderCategory, price} in orders 
+            group by orderCategory where orderCategory == "A" order by orderCategory limit 1 select sum(price);
```
Comment (gimantha) [inline test/resources/config/query.bal:37]: Shall we add this one also for tests?

`var _  = from var {orderCategory, price} in orders 
            group by var x = orderCategory where x == "A" order by x limit 1 select sum(price);`

`orderCategory` is referred by another variable here.

Comment (LakshanWeerasinghe) [reply test/resources/config/query.bal:37]: Fixed.

**Review by gimantha — APPROVED** [review]
LGTM

**Comment by SasinduDilshara** [comment]
> Screenshot 2023-05-17 at 14 35 27 Screenshot 2023-05-17 at 14 37 38

LGTM

**Pull request merged**